### ISSUE TITLE
Meaning of fcMedian in patternTest

### ISSUE THREAD

**Issue by Jabbath** [statOmics/tradeSeq#152]
Hi, I'm using your code to compare trajectories between a normal dataset and a mutant dataset that has some genes knocked out. I'm a little bit confused on what the fcMedian output of patternTest actually means. 

I made a simple example with two sets of cells, where the cells have constant expression over time. The second set of cells had the first 50 genes up-regulated 1.9x compared to the first. fcMedian was log_e(1.9), so I assumed fcMedian is roughly equal to log_e(expression in second trajectory / expression in first). 

However, when I run patternTest on my real dataset, using this definition would suggest that a gene that was knocked out in the mutant is actually upregulated in the mutant. So, I'm wondering what the value of fcMedian actually represents and how it's calculated? More specifically, how can I tell the directionality?

**Comment by HectorRDB** [comment]
Hi @Jabbath 
Thanks for using tradeSeq! I assume you're referring to the code of the workshop: https://kstreet13.github.io/bioc2020trajectories/articles/workshopTrajectories.html

First, I think there is some confusion on the tests. The `patternTest` compares multiple lineages, averaging the results across conditions. The `conditionTest` compares multiple conditions, either at the lineage level, or averaging the results across lineages.

For both those tests, let's say we compare two curves: those tests do a point-by-point comparison (you can refer to the help files of those tests for more details on how those points are selected). The fcMedian is the median fold change between all those points by points comparison. 

It might not always be fully informative, if for example, a gene is first up-regulated in one condition and then later down-regulated, the fcMedian will be close to 1 even though there is some differences.

Let me know if that helps

**Comment by Jabbath** [comment]
Ok that makes sense since then some low-level differences along the curve might lead to unintuitive results. Just to clarify though, for a single point along the curve the fold change is lineage2/lineage1 right?

I think I have been misunderstanding the tests somewhat though. I haven't actually been specifying the conditions. I've been comparing conditions as if they are two separate lineages in one dataset. That is, lineage1 and lineage2 are the same celltype but belong to different conditions. My cellWeights looks something like this:

```
 cellWeights[rownames(wt_endo), "Wild Type"] <- 1
 cellWeights[rownames(mutant_endo), "Mutant"] <- 1 
```
Since I don't actually specify conditions, I assume it's just comparing them as if they were two lineages in a single condition? Is this equivalent to the conditionTest?

**Comment by HectorRDB** [comment]
Yes the fold change is lineage2 / lineage 1 (so Mutant / WT in your case). 

What you did is correct as well, if you only have one lineage per condition, it will give the same results.

**Comment by Jabbath** [comment]
Ok that would make sense. There is still one point of confusion about how I got log(fc) with my example dataset then. Here is the counts matrix I used:
<img src="https://user-images.githubusercontent.com/8083827/120699104-d5f04880-c464-11eb-9d12-fc95670f2891.PNG" height=400px>
The first 2000 columns are "cells" from the first dataset and the last 2000 are from the second dataset. I made 100 timepoints, equally assigning 20 cells from each set to each timepoint. The TMM factors for this dataset are all 1 and I don't do any custom normalization. Here is the output for patternTest:
<img src="https://user-images.githubusercontent.com/8083827/120700483-9296d980-c466-11eb-9d80-18924314f961.PNG" height=400px>
The first 10 genes are 0.642 = ln(19/10) = ln(set2/set1). The last 10 genes are 2.3 = ln(10/1) = ln(set1/set2). My confusion here is about why the logarithm is present, and about why the directionality seems to switch.
